src/test/run-make/save-analysis/foo.rs

```diff
@@ -57,6 +57,12 @@ fn test_alias<I: Iterator>(i: Option<<I as Iterator>::Item>) {
     let y = x.1;
 }
 
+// Issue #37700
+const LUT_BITS: usize = 3;
+pub struct HuffmanTable {
+    ac_lut: Option<[(i16, u8); 1 << LUT_BITS]>,
```

Comment:
I don't really get what the logic is above, but I am wondering if it is going to work for something like `Option<[u8; foo(X > Y)]>` (where `foo()` is a `const fn`)?